server/src/main/java/org/elasticsearch/search/aggregations/bucket/DeferableBucketAggregator.java

```diff
@@ -30,10 +30,12 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
+import java.util.function.BiConsumer;
 
 public abstract class DeferableBucketAggregator extends BucketsAggregator {
 
     private DeferringBucketCollector recordingWrapper;
+    private List<String> deferredAggregationNames;
```

Comment:
Is there not a way to fetch the names out of `collectors` later when we need them?

Reply:
Not without some weird casting! I'll have another look just in case.

Reply:
Ah ok, no worries then :)